src/Surfnet/StepupMiddleware/ApiBundle/Configuration/EventListener/WhitelistEventListener.php

```diff
@@ -0,0 +1,117 @@
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Configuration\Command\CreateInstitutionConfigurationCommand;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\Pipeline;
+
+final class WhitelistEventListener implements EventListenerInterface
+{
+    /**
+     * @var ConfiguredInstitutionRepository
+     */
+    private $configuredInstitutionRepository;
+
+    /**
+     * @var Pipeline
+     */
+    private $pipeline;
+
+    public function __construct(ConfiguredInstitutionRepository $configuredInstitutionRepository, Pipeline $pipeline)
+    {
+        $this->configuredInstitutionRepository = $configuredInstitutionRepository;
+        $this->pipeline                        = $pipeline;
+    }
+
+    /**
+     * @param DomainMessage $domainMessage
+     */
+    public function handle(DomainMessage $domainMessage)
+    {
+        $event  = $domainMessage->getPayload();
+
+        $classParts = explode('\\', get_class($event));
+        $method = 'apply' . end($classParts);
+
+        if (!method_exists($this, $method)) {
+            return;
+        }
+
+        $this->$method($event, $domainMessage);
+    }
+
+    public function applyWhitelistCreatedEvent(WhitelistCreatedEvent $event)
+    {
+        foreach ($event->whitelistedInstitutions as $whitelistedInstitution) {
+            $institution = new Institution($whitelistedInstitution->getInstitution());
+
+            if ($this->configuredInstitutionRepository->hasConfigurationFor($institution)) {
+                continue;
+            }
+
+            $this->createInstitutionConfigurationFor($institution);
+        }
+    }
+
+    public function applyWhitelistReplacedEvent(WhitelistReplacedEvent $event)
+    {
+        foreach ($event->whitelistedInstitutions as $whitelistedInstitution) {
+            $institution = new Institution($whitelistedInstitution->getInstitution());
+
+            if ($this->configuredInstitutionRepository->hasConfigurationFor($institution)) {
+                continue;
+            }
+
+            $this->createInstitutionConfigurationFor($institution);
+        }
+    }
+
+    public function applyInstitutionsAddedToWhitelistEvent(InstitutionsAddedToWhitelistEvent $event)
+    {
+        foreach ($event->addedInstitutions as $addedInstitution) {
+            $institution = new Institution($addedInstitution->getInstitution());
+
+            if ($this->configuredInstitutionRepository->hasConfigurationFor($institution)) {
+                continue;
+            }
+
+            $this->createInstitutionConfigurationFor($institution);
+        }
+    }
+
+    /**
+     * @param Institution $institution
+     */
+    private function createInstitutionConfigurationFor(Institution $institution)
+    {
+        $command              = new CreateInstitutionConfigurationCommand();
+        $command->UUID        = (string) Uuid::uuid4();
+        $command->institution = $institution;
+
+        $this->pipeline->process($command);
```

Comment:
I agree, we would like something intermediary as we're operating on the borders between Event and Command. An added benefit is that this object can be reused wherever we want InstitutionConfigurations to be created. 

I think calling it a factory would be a bit atypical, though, as we are doing more than creating InstitutionConfiguration objects, we are also persisting them. 

I'm not sure, but we could call it a dispatcher as it translates methods (`createFor`) to commands and dispatches them.